tests/Uri/Retrievers/FileGetContentsTest.php

```diff
@@ -59,6 +59,15 @@ public function testContentType()
             $this->assertTrue($fetchContentType->invoke($res, array('Content-Type: application/json')));
             $this->assertFalse($fetchContentType->invoke($res, array('X-Some-Header: whateverValue')));
         }
+
+        public function testCanHandleHttp301PermanentRedirect()
+        {
+            $res = new FileGetContents();
+
+            $res->retrieve('http://asyncapi.com/definitions/2.0.0/asyncapi.json');
```

Comment:
Something like this is, I think, to be avoided. We don't have control over that website or server configuration and it could easily break. Can we perform the same test by mocking a response and not requiring a real external network call? Tests should ideally run without an internet connection.

Reply:
Yes I agree with your point to some extend. In order to test we actually need a schema which is available using an HTTP 301 redirect. My point for using this specific URL is because it is relative safe as it belongs to an organisation and we point to their definition which would not change quickly.